client/verta/tests/unit_tests/conftest.py

```diff
@@ -64,15 +71,137 @@ def __repr__(self):  # avoid network calls when displaying test results
 
 
 @pytest.fixture(scope="session")
-def mock_registered_model_version(mock_conn, mock_config):
-    """Return a mocked object of the RegisteredModelVersion class for use in tests"""
+def mock_simple_pipeline_definition() -> Dict[str, Any]:
+    """Return a mocked pipeline definition for use in tests"""
+
+    def simple_pipeline_definition(id: int) -> Dict[str, Any]:
+        return {
+            "graph": [
+                {"predecessors": [], "name": "step1"},
+                {"predecessors": ["step1"], "name": "step2"},
+            ],
+            "pipeline_version_id": id,
+            "steps": [
+                {
+                    "model_version_id": 1,
+                    "name": "step1",
+                },
+                {
+                    "model_version_id": 2,
+                    "name": "step2",
+                },
+            ],
+        }
+
+    return simple_pipeline_definition
+
+
+@pytest.fixture(scope="session")
+def make_mock_registered_model_version(
+    mock_conn, mock_config, mock_simple_pipeline_definition
+) -> Callable:
+    """Return a callable function for creating mocked objects of the
+    RegisteredModelVersion class for use in tests that require multiple
+    unique instances.
+    """
 
     class MockRegisteredModelVersion(RegisteredModelVersion):
         def __repr__(self):  # avoid network calls when displaying test results
             return object.__repr__(self)
 
-    return MockRegisteredModelVersion(
-        mock_conn,
-        mock_config,
-        _RegistryService.ModelVersion(id=555, registered_model_id=123),
-    )
+        def _get_artifact(self, key=None, artifact_type=None):
+            return json.dumps(mock_simple_pipeline_definition(id=self.id)).encode(
+                "utf-8"
+            )
```

Comment:
This isn't _great_ because other tests may need to mock RMVs but don't necessarily want all artifacts to be pipelines.

Might be fine if there's no other way to do it, though 🤔

Reply:
Actually, I see we have a tests that mocks the HTTP calls for logging the artifact; could we mock getting the artifact, too?

Reply:
I now have a test for getting the artifact, and I have added a conditional statement for the key so that this could easily be extended for other types of artifacts.